lib/package.py

```diff
@@ -161,6 +161,9 @@ def _download_source(self, build_dir):
         An alternative to just execute a given command to obtain sources.
         """
         utils.run_command(self.download_source, cwd=build_dir)
+        # automatically append tar.gz if expects_source has no extension
+        if self.expects_source == self.name:
+            self.expects_source = "%s.tar.gz" % self.name
```

Comment:
GitRepository.archive() and Package._download_source() are mutually exclusive. This patch only alters _download_source(), which uses commands in download_source attribute in package yaml to download package source code. 

The only packages which use this attribute are gcc, libguestfs and kubernetes. This patch was not needed before because the expects_source attribute hardcoded in the packages YAML contained the tar.* extension. The expects_source attribute is now set to be the package name at Package constructor by default and therefore the extension must be added at some point when the default value is used.

An alternative would be to add the tar.gz extension at the constructor, but then other part of the code which clones from a repository instead of allowing custom download commands would need to be updated.